source/connection-troubleshooting.txt

```diff
@@ -0,0 +1,308 @@
+.. _csharp-connection-troubleshooting:
+
+==========================
+Connection Troubleshooting
+==========================
+
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 2
+   :class: singlecol
+
+This page offers potential solutions to issues you might encounter when
+using the {+driver-long+} to connect to a MongoDB deployment.
+
+.. note::
+
+   This page addresses only connection issues. If you encounter any other issues
+   with MongoDB or the driver, visit the following resources:
+
+   - The :ref:`Frequently Asked Questions (FAQ) <csharp-faq>` for the
+     {+driver-short+}
+   - The :ref:`Issues & Help <csharp-issues-help>` page, which has
+     information about reporting bugs, contributing to the driver, and 
+     finding additional resources
+   - The `MongoDB Community Forums <https://community.mongodb.com>`__ for
+     questions, discussions, or general technical support
+
+Connection Error
+~~~~~~~~~~~~~~~~
+
+The following error message indicates that the driver cannot connect to a server
+on the specified hostname or port. Multiple situations can generate this error
+message. In this sample error message, the hostname is ``127.0.0.1`` and the
+port is ``27017``:
+
+.. code-block:: none
+   :copyable: false
+
+   Error: couldn't connect to server 127.0.0.1:27017
+
+The following sections describe actions you can take to potentially resolve the
+issue.
+
+.. _csharp-troubleshooting-connection-string-port:
+
+Check Your Connection String
+----------------------------
+
+Verify that the hostname and port number in the connection string are both
+accurate. The default port value for a MongoDB instance is
+``27017``, but you can configure MongoDB to communicate on another port.
+
+.. _csharp-troubleshooting-connection-firewall:
+
+Configure Your Firewall
+-----------------------
+
+Verify that the ports your MongoDB deployment listens on are not blocked by a
+firewall on the same network. MongoDB uses port ``27017`` by default. To learn
+more about the default ports MongoDB uses and how to change them, see
+:manual:`Default MongoDB Port </reference/default-mongodb-port/>`.
+
+.. warning::
+
+   Do not open a port in your firewall unless you are sure it's the port
+   used by your MongoDB deployment.
+
+Authentication Error
+~~~~~~~~~~~~~~~~~~~~
+
+The {+driver-short+} can fail to connect to a MongoDB instance if
+the authorization is not configured correctly. If you are using ``SCRAM-SHA-256``
```

Comment:
Q: in general, do we use authorization and authentication somewhat interchangeably?

Follow-up S depending on your answer:
```suggestion
the authentication mechanism is not configured correctly. If you are using ``SCRAM-SHA-256``
```